src/EditorFeatures/Core/Implementation/CodeCleanup/CodeCleanupService.cs

```diff
@@ -0,0 +1,165 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Options;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Internal.Log;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.OrganizeImports;
+using Microsoft.CodeAnalysis.RemoveUnnecessaryImports;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.CodeCleanup
+{
+    [Export(typeof(ICodeCleanupService))]
+    internal class CodeCleanupService : ICodeCleanupService
+    {
+        /// <summary>
+        /// Maps format document code cleanup options to DiagnosticId[]
+        /// </summary>
+        private static ImmutableDictionary<PerLanguageOption<bool>, string[]> _dictionary = GetCodeCleanupOptionMapping();
```

Comment:
1. not a huge fan of this being an immutable mapping to mutable data.  Consider using an ```ImmutableArray<string>``` instead of a ```string[]```.
2. _dictionary is too vague a name.

Reply:
fixed